src/main/java/in/org/projecteka/hdaf/link/link/Link.java

```diff
@@ -25,19 +28,23 @@ public Link(HIPClient hipClient, ClientRegistryClient clientRegistryClient) {
         //providerid to be fetched from DB using transactionID
         String providerId = "10000005";
 
+        Patient patientInHIP = toHIPPatient(patientId, patientLinkReferenceRequest);
+        in.org.projecteka.hdaf.link.link.model.hip.PatientLinkReferenceRequest patientLinkReferenceRequestHIP =
+                new in.org.projecteka.hdaf.link.link.model.hip.PatientLinkReferenceRequest(
+                        patientLinkReferenceRequest.getTransactionId(), patientInHIP);
         return providerUrl(providerId)
-                .flatMap(s -> s.map(url -> hipClient.linkPatientCareContext(patientId, patientLinkReferenceRequest, url))
+                .flatMap(s -> s.map(url -> hipClient.linkPatientCareContext(patientLinkReferenceRequestHIP, url))
```

Comment:
Can we please stop having suffix `HIP` on the variable names, please please?

Reply:
`
    
        return providerUrl(providerId)
                .flatMap(url -> hipClient.linkPatientCareContext(patientId, patientLinkReferenceRequest, url))
                .switchIfEmpty(Mono.error(new Throwable("Invalid HIP")));
    }

    public Mono<PatientLinkResponse> verifyToken(String patientId, String linkRefNumber, PatientLinkRequest patientLinkRequest) {
        //from linkRefNumber get TransactionId
        //from transactionID get providerID
        String providerId = "10000005";
        //Check otp for expiry
        return providerUrl(providerId)
                .flatMap(url -> hipClient.validateToken(patientId, linkRefNumber, patientLinkRequest, url))
                .switchIfEmpty(Mono.error(new Throwable("Invalid HIP")));
    }

    private Mono<String> providerUrl(String providerId) {
        return clientRegistryClient.providerWith(providerId)
                .flatMap(provider -> provider.getIdentifiers()
                        .stream()
                        .findFirst()
                        .map(identifier -> Mono.just(identifier.getSystem()))
                        .orElse(Mono.empty()));
    }`